embedded-pubsub/src/main/java/com/playtika/test/pubsub/PubSubResourcesGenerator.java

```diff
@@ -0,0 +1,104 @@
+package com.playtika.test.pubsub;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.grpc.GrpcTransportChannel;
+import com.google.api.gax.rpc.AlreadyExistsException;
+import com.google.api.gax.rpc.FixedTransportChannelProvider;
+import com.google.api.gax.rpc.TransportChannelProvider;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.cloud.pubsub.v1.SubscriptionAdminClient;
+import com.google.cloud.pubsub.v1.SubscriptionAdminSettings;
+import com.google.cloud.pubsub.v1.TopicAdminClient;
+import com.google.cloud.pubsub.v1.TopicAdminSettings;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.ProjectTopicName;
+import com.google.pubsub.v1.PushConfig;
+import com.google.pubsub.v1.Subscription;
+import com.google.pubsub.v1.Topic;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import org.testcontainers.containers.GenericContainer;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+
+public class PubSubResourcesGenerator {
```

Comment:
https://github.com/GoogleCloudPlatform/cloud-shell-tutorials/blob/master/cloud-console-tutorials/pubsub_quickstart/pubsub_quickstart.md#add-a-subscription

if you exec in docker container - no need for java libs in classpath which might conflict with versions of java sdk used by service. With kafka client it was huge issue.
https://github.com/testcontainers/testcontainers-spring-boot/blob/master/embedded-kafka/src/main/java/com/playtika/test/kafka/KafkaTopicsConfigurer.java

Reply:
@renanreismartins don't want to switch to CLI and make all deps optional?